server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAO.java

```diff
@@ -108,7 +108,7 @@ public void close() {
             .contentLength(payload.getLength().orElse(Long.valueOf(data.length)))
             .build();
 
-        return Mono.fromRunnable(() -> blobPutter.putDirectly(resolvedBucketName, blob))
+        return blobPutter.putDirectly(resolvedBucketName, blob)
```

Comment:
I think that this  `save` method is lying anyway because it does non-trivial things before returning the Mono.
Would you mind putting everything in the Mono?